components/eamxx/src/diagnostics/field_at_height.cpp

```diff
@@ -38,6 +38,12 @@ FieldAtHeight (const ekat::Comm& comm, const ekat::ParameterList& params)
  : AtmosphereDiagnostic(comm,params)
 {
   m_field_name = m_params.get<std::string>("field_name");
+  auto surf_ref = m_params.get<std::string>("surface_reference");
+  EKAT_REQUIRE_MSG(surf_ref == "sealevel" or surf_ref == "surface",
+      "Error! Invalid surface reference for FieldAtHeight.\n"
+      " -        field name: " + m_field_name + "\n"
+      " - surface reference: " + surf_ref + "\n");
+  m_surf_ref = (surf_ref == "sealevel") ? "z" : "geopotential";
```

Comment:
This is minor, but I think you are abusing the term "surf_ref" here. Perhaps you could use `m_vert_coord_name`, since you are storing the coord name, rather than the surface reference name...

Reply:
Although, that may seem confusingly similar to `m_z_name`. Perhaps you should store "sealevel"/"surface" in `m_surf_name`, and do a switch on it when you need it, like

```
m_z_name = (m_surf_ref=="sealevel" ? "z" : "geopotential") + (tag==LEV" ? "_mid" : "_int");
```

Reply:
Another possibility: do away with `m_surf_ref`, and instead:
- set `m_z_name` to either `z` or `geopotential`
- store `m_z_suffix` to be either `_mid` or `_int`.

Then you can do stuff like
```
add_field<Required>(m_z_name+m_z_suffix,gname);
```
or
```
const auto z_view = get_field_in(m_z_name+m_z_suffix).get_view<const Real**>();
```